examples/diff/index.html

```diff
@@ -0,0 +1,160 @@
+  <div class="cf-form-hint">
+    <span class="diff-text"></span>
+  </div>
+</div>
+
+<!-- Contenful UI Extensions SDK API -->
+<script src="//contentful.github.io/ui-extensions-sdk/cf-widget-api.js"></script>
+<!-- Diff Lib -->
+<script src="//cdnjs.cloudflare.com/ajax/libs/jsdiff/2.2.2/diff.min.js"></script>
+
+<!-- Diffing UI Extension -->
+<script>
+
+  function diffExtension(extensionsApi) {
+
+    extensionsApi.window.startAutoResizer();
+
+    // DOM elements.
+    var draftTextEl = document.querySelector('.draft-text');
+    var diffTextEl = document.querySelector('.diff-text');
+
+    // Disable the input as we need to wait for an async call to load the published entry.
+    draftTextEl.disabled = true;
+
+    var entry = extensionsApi.entry;
+    var field = extensionsApi.field;
+
+    // Currently published version of entry.
+    var publishedVersion = extensionsApi.entry.getSys().publishedVersion;
+    var publishedValue;
+
+    // Current locale
+    var fieldLocale = extensionsApi.field.locale;
+    var fieldId = extensionsApi.field.id;
+
+    // Callbacks for changes on the field or the entry.sys object.
+    // Detach functions could be used to clean up the callbacks.
+    var detachValueChanged = field.onValueChanged(onValueChanged);
+    var detachSysChanged = entry.onSysChanged(onSysChanged);
+
+    draftTextEl.addEventListener('input', keyboardInputHandler);
+
+    if (publishedVersion) {
+      // Load published value from CDA.
+      loadPublishedValue(entry.getSys().id)
+          .then(function (entries) {
+                publishedValue = entries.items[0].fields[fieldId][fieldLocale];
+                draftTextEl.disabled = false;
+                renderDiff();
+              }
+          )
+          .catch(function (error) {
+            console.log(error)
+          });
+    } else {
+      draftTextEl.disabled = false;
+      renderDiff();
+    }
+
+    function keyboardInputHandler() {
+      renderDiff();
+      field.setValue(draftTextEl.value);
+    }
+
+    /**
+     * Gets the published value from the API.
+     * @param entyId The id of the entry to retrieve.
+     * @return Promise A Promise object of the call.
+     */
+    function loadPublishedValue(entryId) {
+      return extensionsApi.space.getPublishedEntries({'sys.id': entryId})
+    }
+
+    /**
+     * Uses diff lib to render diff to DOM of extension.
+     */
+    function renderDiff() {
+
+      // Do not render if there is no published value.
+      if (!publishedVersion) {
+        diffTextEl.innerHTML = '<i>Entry not yet published.</i>';
+        return;
+      }
+
+      // Do not render if there is no diff.
+      if (publishedValue === draftTextEl.value) {
+        diffTextEl.innerHTML = '<i>Draft and published value are equal.</i>';
+        return;
+      }
+
+      // Clear diff DOM element.
+      diffTextEl.innerHTML = '';
+
+      var diff = JsDiff.diffWordsWithSpace(publishedValue, draftTextEl.value);
+
+      var className;
```

Comment:
I would move `var className` and `var spanEl` inside the function passed to `forEach` since they are only used in there.